File: examples/c++/adafruitms1438-stepper.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>

#include "adafruitms1438.hpp"
#include "upm_utilities.h"

using namespace std;
using namespace upm;

int
main(int argc, char** argv)
{
    //! [Interesting]
    // Instantiate an Adafruit MS 1438 on I2C bus 0

    upm::AdafruitMS1438 ms(ADAFRUITMS1438_I2C_BUS, ADAFRUITMS1438_DEFAULT_I2C_ADDR);

    // Setup for use with a stepper motor connected to the M1 & M2 ports

    // set a PWM period of 50Hz

    // disable first, to be safe
    ms.disableStepper(AdafruitMS1438::STEPMOTOR_M12);

    // configure for a NEMA-17, 200 steps per revolution
    ms.stepConfig(AdafruitMS1438::STEPMOTOR_M12, 200);

    // set speed at 10 RPM's
    ms.setStepperSpeed(AdafruitMS1438::STEPMOTOR_M12, 10);
    ms.setStepperDirection(AdafruitMS1438::STEPMOTOR_M12, AdafruitMS1438::DIR_CW);

    // enable
    cout << "Enabling..." << endl;
    ms.enableStepper(AdafruitMS1438::STEPMOTOR_M12);

    cout << "Rotating 1 full revolution at 10 RPM speed." << endl;
    ms.stepperSteps(AdafruitMS1438::STEPMOTOR_M12, 200);

    cout << "Sleeping for 2 seconds..." << endl;
    upm_delay(2);
    cout << "Rotating 1/2 revolution in opposite direction at 10 RPM speed." << endl;

    ms.setStepperDirection(AdafruitMS1438::STEPMOTOR_M12, AdafruitMS1438::DIR_CCW);
    ms.stepperSteps(AdafruitMS1438::STEPMOTOR_M12, 100);

    cout << "Disabling..." << endl;
    ms.disableStepper(AdafruitMS1438::STEPMOTOR_M12);

    cout << "Exiting" << endl;

    //! [Interesting]

    return 0;
}

File: examples/javascript/adafruitms1438-stepper.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

function exit()
{
	console.log("Exiting");

	myMotorShield_obj = null;
	if (MotorShield_lib)
	{
		MotorShield_lib.cleanUp();
		MotorShield_lib = null;
	}
	process.exit(0);
}

var MotorShield_lib = require('jsupm_adafruitms1438');

/* Import header values */
var I2CBus = MotorShield_lib.ADAFRUITMS1438_I2C_BUS;
var I2CAddr = MotorShield_lib.ADAFRUITMS1438_DEFAULT_I2C_ADDR;

var M12motor = MotorShield_lib.AdafruitMS1438.STEPMOTOR_M12;
var MotorDirCW = MotorShield_lib.AdafruitMS1438.DIR_CW;
var MotorDirCCW = MotorShield_lib.AdafruitMS1438.DIR_CCW;


// Instantiate an Adafruit MS 1438 on I2C bus 0
var myMotorShield_obj = new MotorShield_lib.AdafruitMS1438(I2CBus, I2CAddr);


// Setup for use with a stepper motor connected to the M1 & M2 ports

// disable first, to be safe
myMotorShield_obj.disableStepper(M12motor);

// configure for a NEMA-17, 200 steps per revolution
myMotorShield_obj.stepConfig(M12motor, 200);

// set speed at 10 RPM's
myMotorShield_obj.setStepperSpeed(M12motor, 10);
myMotorShield_obj.setStepperDirection(M12motor, MotorDirCW);

console.log("Enabling...");
myMotorShield_obj.enableStepper(M12motor);

console.log("Rotating 1 full revolution at 10 RPM speed.");
myMotorShield_obj.stepperSteps(M12motor, 200);

console.log("Sleeping for 2 seconds...");


setTimeout(function()
{
	console.log("Rotating 1/2 revolution in opposite direction at 10 RPM speed.");

	myMotorShield_obj.setStepperDirection(M12motor, MotorDirCCW);
	myMotorShield_obj.stepperSteps(M12motor, 100);

	console.log("Disabling...");
	myMotorShield_obj.disableStepper(M12motor);
	exit();
}, 2000);


process.on('SIGINT', function()
{
	exit();
});
